File: app/src/main/java/eungju/rxcurrencyconverter/CurrencyApplication.java
package eungju.rxcurrencyconverter;

import android.app.Application;
import android.content.Context;

import timber.log.Timber;

public class CurrencyApplication extends Application {
    private ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }

        component = ApplicationComponent.Builder.build(this);
        component.inject(this);
    }

    public ApplicationComponent component() {
        return component;
    }

    public static CurrencyApplication get(Context context) {
        return (CurrencyApplication) context.getApplicationContext();
    }
}

File: app/src/main/java/eungju/rxcurrencyconverter/CurrencyApplication.kt
package eungju.rxcurrencyconverter

import android.app.Application
import android.content.Context

import timber.log.Timber

class CurrencyApplication : Application() {
    private lateinit var component: ApplicationComponent

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }

        component = ApplicationComponent.Builder.build(this)
        component.inject(this)
    }

    fun component(): ApplicationComponent {
        return component
    }

    companion object {
        operator fun get(context: Context): CurrencyApplication {
            return context.applicationContext as CurrencyApplication
        }
    }
}
